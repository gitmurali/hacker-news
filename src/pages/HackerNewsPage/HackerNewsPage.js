import React, { useEffect, useState } from 'react'

import HackerNewsHeader from '@components/HackerNewsHeader'
import NewsFeeds from '@components/NewsFeeds'
import Page from '@components/Page'

import styles from './HackerNewsPage.scss'

const HackerNewsPage = ({ hackerNews, onload, handleMore }) => {
  const [pages, setPages] = useState(0)
  const [counter, setCounter] = useState(0)

  useEffect(() => {
    onload()
  }, [])

  useEffect(() => {
    if (hackerNews.nbPages) setPages(hackerNews.nbPages)
  }, [hackerNews.nbPages])

  useEffect(() => {
    if (counter !== 0 && counter + 1 <= pages) {
      handleMore(counter)
    }
  }, [counter])

  return (
    <div className={styles.marginTop}>
      <Page contentCentered>
        <HackerNewsHeader />
        <NewsFeeds hackerNews={hackerNews} />
        <div className={styles.more} onClick={() => setCounter(counter + 1)}>
          More
        </div>
      </Page>
    </div>
  )
}

export default HackerNewsPage
