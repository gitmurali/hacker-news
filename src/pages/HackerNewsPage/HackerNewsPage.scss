.margin-top {
  margin-top: 25px;
}
.more {
  font-size: 14px;
  color: $orange;
  margin-top: 10px;
  cursor: pointer;
}
