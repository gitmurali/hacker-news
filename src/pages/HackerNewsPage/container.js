import { getHackerNewsRequest } from '@actions/hackerNews'
import HackerNewsPage from '@pages/HackerNewsPage/HackerNewsPage'
import { connect } from 'react-redux'

const mapStateToProps = ({ hackerNews }) => ({
  hackerNews
})

const mergeProps = (state, { dispatch }, props) => ({
  ...state,
  ...props,
  onload: () => {
    dispatch(getHackerNewsRequest({ page: 0 }))
  },
  handleMore: page => {
    dispatch(getHackerNewsRequest({ page }))
  }
})

export default connect(mapStateToProps, null, mergeProps)(HackerNewsPage)
