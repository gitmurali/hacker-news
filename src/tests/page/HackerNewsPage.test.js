import React from 'react'

import HackerNewsPage from '@pages/HackerNewsPage'
import ConnectedIntlProvider from '@utils/IntlConnect/IntlConnect'
import { shallow, mount } from 'enzyme'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import configureMockStore from 'redux-mock-store'

const mockStore = configureMockStore()
describe('Component:HackerNewsPage', () => {
  let store
  beforeEach(() => {
    store = mockStore({})
  })
  test('should match snapshot', () => {
    const wrapper = shallow(
      <IntlProvider locale='en'>
        <HackerNewsPage />
      </IntlProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('should render feeds', () => {
    const temp = {
      hackerNews: {
        hits: []
      }
    }
    const noop = () => {}
    const wrapper = mount(
      <Provider store={store}>
        <ConnectedIntlProvider>
          <BrowserRouter>
            <HackerNewsPage hackerNews={temp} onload={noop} handleMore={noop}/>
          </BrowserRouter>
        </ConnectedIntlProvider>
      </Provider>
    )
    expect(wrapper.find('NewsFeeds')).toHaveLength(1)
  })
})
