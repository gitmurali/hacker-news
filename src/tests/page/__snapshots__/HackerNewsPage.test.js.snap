// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:HackerNewsPage should match snapshot 1`] = `<Connect(HackerNewsPage) />`;
